test/rendering.test.js

```diff
@@ -496,198 +496,139 @@ describe('Rendering', () => {
     });
   });
 
-  describe('Miscellaneous', () => {
-    it('sets the surrogate-keys correctly', async () => {
-      const resp = await testRender('page-block-empty-cols');
-      assert.strictEqual(resp.headers.get('x-surrogate-key'), 'rDFj9gBeGHx_FI2T foo-id_metadata super-test--helix-pages--adobe_head foo-id');
-    });
-
-    it('sets the surrogate-keys correctly for plain', async () => {
-      const resp = await testRenderPlain('one-section');
-      assert.strictEqual(resp.headers.get('x-surrogate-key'), 'oHjg_WDu20CBS4rD foo-id_metadata super-test--helix-pages--adobe_head foo-id');
-    });
-
-    it('sets the surrogate-keys correctly for index.plain.html', async () => {
-      const resp = await testRenderPlain('one-section/index', 'one-section/index');
-      assert.strictEqual(resp.headers.get('x-surrogate-key'), 'Vp-I6NB8PSor1sI6 foo-id_metadata super-test--helix-pages--adobe_head foo-id');
-    });
-
-    it('renders the fedpub header correctly', async () => {
-      await testRenderPlain('fedpub-header');
-    });
-
-    it('renders styling test document correctly', async () => {
-      await testRenderPlain('styling');
-    });
+  describe('csp', () => {
+    const originalRandomUUID = cryptoImpl.randomUUID;
+    const uuids = [
+      '7241346e-6430-6d6d-6d72-41346e64306d',
+      '6d6d0000-0000-0000-0000-000000000000',
+    ];
+    let uuidIdx = 0;
 
-    it('renders document with gridtables correctly', async () => {
-      await testRender('page-with-gridtables');
+    beforeEach(() => {
+      uuidIdx = 0;
+      // eslint-disable-next-line no-plusplus
+      cryptoImpl.randomUUID = () => uuids[uuidIdx++];
     });
 
-    it('renders document with many image references quickly', async () => {
-      await testRender('gt-many-refs');
+    afterEach(() => {
+      cryptoImpl.randomUUID = originalRandomUUID;
     });
```

Comment:
I like this better than esmock...

Reply:
Let's discuss here: https://github.com/adobe/helix-html-pipeline/pull/817#discussion_r1954758978
Yes, the v2 is simply because `randomUUID` is easier to mock, but makes the implementation of the actual nonce generation a little more complex.